docs/docs/view-config-js.md

```diff
@@ -0,0 +1,474 @@
+
+
+## `VitessceConfig`
+
+`VitessceConfig` is a class representing a Vitessce view config. To begin creating a view config with the API, you will need to instantiate a `VitessceConfig` object.
+The methods of this object (and the objects its methods return) allow you to manipulate the underlying configuration.
+When you are ready, you will use the `.toJSON()` method to convert the `VitessceConfig` object to a plain JSON object.
+
+
+### `constructor(name, description)`
+
+Construct a Vitessce view config object.
+
+
+#### Parameters:
+- `name` (`string`) - A name for the view config.
+- `description` (`string`) - A description for the view config. Optional.
+
+```js {3}
+import { VitessceConfig } from 'vitessce';
+
+const vc = new VitessceConfig("My config");
+```
+
+
+### `addDataset(name)`
+
+Add a dataset to the config.
+
+#### Parameters:
+- `name` (`string`) - A name for the dataset.
+
+#### Returns:
+- Type: `VitessceConfigDataset`
+
+Returns the instance for the new dataset.
+
+
+```js {4}
+import { VitessceConfig } from 'vitessce';
+
+const vc = new VitessceConfig("My config");
+const dataset = vc.addDataset("My dataset")
+    .addFile(
+        "http://example.com/my-cells-data.json",
+        "cells",
+        "cells.json"
+    );
+```
+
+
+### `addView(dataset, component, extra)`
+
+Add a view to the config.
+
+#### Parameters:
+- `dataset` (`VitessceConfigDataset`) - A dataset instance to be used for the data visualized in this view.
+- `component` (`string`) - A component name.
+- `extra` (`object`) - An optional object with extra parameters.
+    - `mapping` (`string`) - A convenience parameter for setting the `embeddingType` coordination scope value. This parameter is only applicable when adding the `scatterplot` component. Optional.
+    - `x` (`number`) - The horizontal position of the view. Must be an integer between 0 and 11. Optional.
+    - `y` (`number`) - The vertical position of the view. Must be an integer between 0 and 11. Optional.
+    - `w` (`number`) - The width of the view. Must be an integer between 1 and 12. Optional.
+    - `h` (`number`) - The height of the view. Must be an integer between 1 and 12. Optional.
+
+#### Returns:
+- Type: `VitessceConfigView`
+
+Returns the instance for the new view.
+
+```js {5-6}
+import { VitessceConfig } from 'vitessce';
+
+const vc = new VitessceConfig("My config");
+const dataset = vc.addDataset("My dataset");
+const v1 = vc.addView(dataset, "spatial");
```

Comment:
Relatedly, this component name (`spatial`) is exported as an object that contains the constants?  We should use that here, and note that it is exported.